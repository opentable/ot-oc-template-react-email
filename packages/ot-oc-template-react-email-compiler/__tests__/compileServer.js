jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;

const path = require("path");
const fs = require("fs-extra");
const compileServer = require("../lib/compileServer.js");

const componentPath = path.join(__dirname, "../../../mocks/react-component");
const publishPath = path.join(componentPath, "_packageCompileServerTest");
const publishFileName = "server.js";

const options = {
  componentPackage: fs.readJsonSync(`${componentPath}/package.json`),
  componentPath,
  publishPath,
  publishFileName,
  production: true
};

test("Should correctly compile the server", done => {
  compileServer(
    Object.assign({}, options, {
      compiledViewInfo: { bundle: { hashKey: "666" } }
    }),
    (err, compiledServerInfo) => {
      expect(err).toBeNull();
      compiledServerInfo.hashKey = "dummyData";
      expect(compiledServerInfo).toMatchSnapshot();
      expect(
        fs.readFileSync(path.join(publishPath, publishFileName), "UTF8")
      ).toMatchSnapshot();
      fs.removeSync(publishPath);
      done();
    }
  );
});
