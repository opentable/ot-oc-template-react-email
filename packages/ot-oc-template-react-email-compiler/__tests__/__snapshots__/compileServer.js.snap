// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the server 1`] = `
Object {
  "hashKey": "dummyData",
  "src": "server.js",
  "type": "node.js",
}
`;

exports[`Should correctly compile the server 2`] = `"module.exports=function(a){function b(d){if(c[d])return c[d].exports;var e=c[d]={i:d,l:!1,exports:{}};return a[d].call(e.exports,e,e.exports,b),e.l=!0,e.exports}var c={};return b.m=a,b.c=c,b.d=function(a,c,d){b.o(a,c)||Object.defineProperty(a,c,{enumerable:!0,get:d})},b.r=function(a){'undefined'!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(a,Symbol.toStringTag,{value:'Module'}),Object.defineProperty(a,'__esModule',{value:!0})},b.t=function(a,c){if(1&c&&(a=b(a)),8&c)return a;if(4&c&&'object'==typeof a&&a&&a.__esModule)return a;var d=Object.create(null);if(b.r(d),Object.defineProperty(d,'default',{enumerable:!0,value:a}),2&c&&'string'!=typeof a)for(var e in a)b.d(d,e,function(b){return a[b]}.bind(null,e));return d},b.n=function(a){var c=a&&a.__esModule?function(){return a['default']}:function(){return a};return b.d(c,'a',c),c},b.o=function(a,b){return Object.prototype.hasOwnProperty.call(a,b)},b.p='',b(b.s=0)}([function(a,b,c){'use strict';c.r(b);const d=(a,b)=>{const c=a.params.name;return b(null,{name:c})};c.d(b,'data',function(){return e});const e=(a,b)=>{d(a,(c,d)=>{if(c)return b(c);const e=Object.assign({},d,{_staticPath:a.staticPath,_baseUrl:a.baseUrl,_componentName:'react-component',_componentVersion:'1.0.0'}),f=0===a.staticPath.indexOf('http'),g=f?a.staticPath:'https:'+a.staticPath;return b(null,Object.assign({},{reactComponent:{key:'666',src:g+'react-component.js',props:e}}))})}}]);"`;
